File: apps/BuddyPlay/shared/BuddyCore/Sources/BuddyCore/Observability/AnalyticsService.swift
import Foundation

/// Telemetry slot. v1 attaches `NoopAnalyticsService`; v1.1 will optionally
/// attach a real PostHog/Mixpanel transport behind a Settings opt-in.
///
/// IMPORTANT: BuddyPlay v1 does NOT send any telemetry. This file declares
/// the interface only — no SDK is imported, no transport is wired.
public protocol AnalyticsService {
    func track(_ event: String, properties: [String: String])
    func screen(_ name: String)
    func setUserProperty(_ key: String, value: String)
}

public final class NoopAnalyticsService: AnalyticsService {
    public init() {}
    public func track(_ event: String, properties: [String: String]) {}
    public func screen(_ name: String) {}
    public func setUserProperty(_ key: String, value: String) {}
}

#if canImport(PostHog)
// Real implementation lands in v1.1 once the user opts in via Settings.
// import PostHog
// public final class PostHogAnalyticsService: AnalyticsService { ... }
#endif

File: apps/BuddyPlay/android/core/src/main/java/com/americangroupllc/buddyplay/core/observability/AnalyticsService.kt
package com.americangroupllc.buddyplay.core.observability

/**
 * Telemetry slot. v1 attaches [NoopAnalyticsService]; v1.1 will optionally
 * attach a real PostHog/Mixpanel transport behind a Settings opt-in.
 *
 * IMPORTANT: BuddyPlay v1 does NOT send any telemetry. This file declares
 * the interface only — no SDK is imported.
 */
interface AnalyticsService {
    fun track(event: String, properties: Map<String, String> = emptyMap())
    fun screen(name: String)
    fun setUserProperty(key: String, value: String)
}

class NoopAnalyticsService : AnalyticsService {
    override fun track(event: String, properties: Map<String, String>) {}
    override fun screen(name: String) {}
    override fun setUserProperty(key: String, value: String) {}
}
